5.x/crud-fields.md

```diff
@@ -466,6 +466,73 @@ Input preview:
 
 <hr>
 
+<a name="google-map"></a>
+### google_map
+
+Uses the Google Places API to generate a Map where user can navigate and select a position to store `Latitude` (**lat**), `Longitude` (**lng**) and the address string (**formatted_address**). By default it saves a string in dabase with `{lat: 123, lng: 456, formatted_address: 'Lisbon, Portugal'}`. 
+**Note**: If you want two save in separate fields continue reading below.
+
+```php
+CRUD::addField([
+    'name' => 'location',
+    'type' => 'google_map',
+    // optionals
+    'map_options' => [
+        'default_lat' => 123,
+        'default_lng' => 456,
+        'locate' => false, // enable/disable the `Get my location`
+        'height' => 400 // in pixels
+    ]
+]);
+```
+
+Using Google Places API is dependent on using an API Key. Please [get an API key](https://console.cloud.google.com/apis/credentials) - you do have to configure billing, but you qualify for $200/mo free usage, which covers most use cases. Then copy-paste that key as your ```services.google_places.key``` value. So inside your ```config/services.php``` please add the items below:
+
+```php
+'google_places' => [
+    'key' => 'the-key-you-got-from-google-places'
+],
+```
+
+### SAVING IN MULTIPLE INPUTS
+There are cases where you rather save the information on separate inputs in the database. 
+In that scenario you should use [Laravel mutators and accessors](https://laravel.com/docs/9.x/eloquent-mutators) .
+
+Using the same field as previously shown (**field name is `location`**), and having `latitude`, `longitude`, `full_address` as the database fields, we can save and retrieve them with the following example:
+```php
+
+//add all the fields to model fillable property, including the one that we are not going to save (location in the example)
+$fillable = ['location', 'latitude', 'longitude', 'full_address'];
+
+// 
+protected function location(): \Illuminate\Database\Eloquent\Casts\Attribute
+{
+    return \Illuminate\Database\Eloquent\Casts\Attribute::make(
+        get: fn ($value, $attributes) => json_encode([
+            'lat' => $attributes['lat'], 
+            'lng' => $attributes['lng'], 
+            'formatted_address' => $attributes['full_address'] ?? ''
+            ],
+             JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT | JSON_THROW_ON_ERROR),
+        set: function($value) {
+            $location = json_decode($value);
+            return [
+                'lat' => $location->lat,
+                'lng' => $location->lng,
+                'full_address' => $location->formatted_address ?? ''
+            ];
+        }
+    );
+}
+
+```
+
+Input preview:
```

Comment:
Let's upload an image for it here, inside the Github conversation. 
Then we can take that URL and introduce it in the markdown 😉